tests/unit/client-eviction.tcl

```diff
@@ -456,11 +460,16 @@ start_server {} {
             for {set j 0} {$j < $clients_per_size} {incr j} {
                 set rr [redis_client]
                 lappend rrs $rr
-                $rr client setname client-$i
+                $rr client setname client-$i-$j
+                wait_for_condition 500 10 {
+                    [client_field client-$i-$j rbs] < [kb 2]
+                } else {
+                    fail "client client-$i-$j failed to shrink reply buffer after 5 seconds"
+                }
```

Comment:
so we'll wait an additional 5 seconds per client?
that's unacceptable.
even waiting 5 seconds after the loop for all clients to be resized (one 5 seconds delay in contrast to one per client) would be too much.
we should either find some trick to avoid the buffer being resized (keep a high peak),
or we add a hidden config for this this shrinking and set to very high value.

Reply:
@oranagra the clients are no expected to wait 5 seconds. I mainly did that in order to provide some upper limit.
the clients are starting idle, so I only want to make sure the cron will shrink their buffers (does not require 5 seconds peak reset)
However I am supportive for some feature disablement mechanism. I offered to support a debug sub command to disable this feature but a config is also possible.

Reply:
ohh, i now realize that setting `buf_peak_last_reset_time` to 0 at client creations means that it's gonna get resized for the first time very quickly, possibly before it manages to send anything.
maybe that's not wise?
maybe we should either set `buf_peak_last_reset_time` to "now" on client creation, or if we set it to 0, we should probably set the startup `c->buf` to `PROTO_REPLY_MIN_BYTES` instead of `PROTO_REPLY_CHUNK_BYTES`.

setting it to MIN_BYTES (and setting `reset_time` to 0) could cause a bigger performance regression, so from performance standpoint, we better keep the big allocation, and use "now".
on the other hand, using MIN_BYTES will make our tests easier... but as noted earlier, test shouldn't influence production decisions.

so please use "now" and add a DEBUG control for this threshold (i think it's slightly better than a hidden config for this purpose).

Reply:
@oranagra I believe that your point is to be:
1. set the initial value of buf_peak_last_reset_time to server.unixtime on client creation
2. set the initial value of buf_peak to PROTO_REPLY_CHUNK_BYTES
So that init clients will not get shrinked too soon.
I think that would be O.K to change.